dev/tests/scheduler.rs

```diff
@@ -346,3 +388,46 @@ async fn test_update_backend_stats_message() {
         result.0
     );
 }
+
+#[integration_test]
+async fn schedule_request_lock() {
```

Comment:
Should we add a test that shows releasing and creating a new lock works as expected?